kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/S3SinkConfig.scala

```diff
@@ -21,32 +21,39 @@ import cats.syntax.all._
 import com.datamountaineer.kcql.Kcql
 import com.typesafe.scalalogging.LazyLogging
 import io.lenses.streamreactor.connect.aws.s3.config.Format.Json
+import io.lenses.streamreactor.connect.aws.s3.config.S3ConfigSettings.{SEEK_MAX_INDEX_FILES, SEEK_MAX_INDEX_FILES_DEFAULT, SEEK_MIGRATION, SEEK_MIGRATION_DEFAULT}
 import io.lenses.streamreactor.connect.aws.s3.config.S3FlushSettings.{defaultFlushCount, defaultFlushInterval, defaultFlushSize}
 import io.lenses.streamreactor.connect.aws.s3.config.{FormatSelection, S3Config, S3ConfigDefBuilder}
 import io.lenses.streamreactor.connect.aws.s3.model.location.RemoteS3RootLocation
 import io.lenses.streamreactor.connect.aws.s3.model.{LocalStagingArea, PartitionSelection}
 import io.lenses.streamreactor.connect.aws.s3.sink._
 
+import scala.util.Try
+
 object S3SinkConfig {
 
-  def apply(s3ConfigDefBuilder: S3ConfigDefBuilder): Either[Exception, S3SinkConfig] = {
-    SinkBucketOptions(s3ConfigDefBuilder) match {
-      case Left(ex) => ex.asLeft[S3SinkConfig]
-      case Right(value) => S3SinkConfig(S3Config(s3ConfigDefBuilder.getParsedValues), value).asRight
-    }
+  def apply(s3ConfigDefBuilder: S3ConfigDefBuilder): Either[Throwable, S3SinkConfig] = {
+    for {
+      sinkBucketOptions <- SinkBucketOptions(s3ConfigDefBuilder)
+      offsetSeekerOptions = OffsetSeekerOptions(
+          Try(s3ConfigDefBuilder.getInt(SEEK_MAX_INDEX_FILES).toInt).getOrElse(SEEK_MAX_INDEX_FILES_DEFAULT),
```

Comment:
if we define a field as boolean/int the Connect API ensures the type so we don't need, technically, to re-validate. Connect will throw an error  if the type is not correct

Reply:
ok, so I will simplify this to 
`offsetSeekerOptions = OffsetSeekerOptions(
        s3ConfigDefBuilder.getInt(SEEK_MAX_INDEX_FILES),
        s3ConfigDefBuilder.getBoolean(SEEK_MIGRATION)
      )`